Do not run changelog check when there are no changes

```
git clone https://github.com/cs3org/reva
cd reva
make deps && make
```

will fail with:

```
go run tools/check-changelog/main.go
2020/12/10 08:52:13 No changelog added. Please create a changelog item based on your changes
exit status 1
make: *** [check-changelog] Error 1
```

--- a/changelog/unreleased/changelog-fix.md
+++ b/changelog/unreleased/changelog-fix.md
@@ -1,0 +1,3 @@
+Bugfix: Run changelog check only if there are changes in the code
+
+https://github.com/cs3org/reva/pull/1385

--- a/tests/acceptance/expected-failures-on-OWNCLOUD-storage.txt
+++ b/tests/acceptance/expected-failures-on-OWNCLOUD-storage.txt
@@ -1456,7 +1456,6 @@ apiWebdavEtagPropagation1/moveFileFolder.feature:62
 apiWebdavEtagPropagation1/moveFileFolder.feature:78
 apiWebdavEtagPropagation1/moveFileFolder.feature:79
 apiWebdavEtagPropagation1/moveFileFolder.feature:98
-apiWebdavEtagPropagation1/moveFileFolder.feature:99
 apiWebdavEtagPropagation1/moveFileFolder.feature:146
 apiWebdavEtagPropagation1/moveFileFolder.feature:147
 apiWebdavEtagPropagation1/moveFileFolder.feature:174

--- a/tools/check-changelog/main.go
+++ b/tools/check-changelog/main.go
@@ -70,8 +70,19 @@ func main() {
 	if *repo != "" {
 		branch = *repo + "/master"
 	}
-	cmd := exec.Command("git", "diff-index", branch, "--", "changelog/unreleased")
+
+	cmd := exec.Command("git", "diff-index", branch, "--", ".")
 	out, err := cmd.Output()
+	if err != nil {
+		log.Fatal(err)
+	}
+	// Return successfully if there are no changes
+	if len(out) == 0 {
+		return
+	}
+
+	cmd = exec.Command("git", "diff-index", branch, "--", "changelog/unreleased")
+	out, err = cmd.Output()
 	if err != nil {
 		log.Fatal(err)
 	}

